File: sample/sample/InstreamViewController.m
//
//  InstreamViewController.m
//  sample
//
//  Created by Vladas Drejeris on 18/05/16.
//  Copyright © 2016 adform. All rights reserved.
//

#import "InstreamViewController.h"
#import <AdformAdvertising/AdformAdvertising.h>
#import <AVKit/AVKit.h>

static NSInteger const kPreRollMid = 142494;
static NSInteger const kMidRollMid = 142494;
static NSInteger const kPostRollMid = 142494;

@interface InstreamViewController ()

@property (nonatomic, strong) IBOutlet AFVideoPlayerController *af_videoPlayer;
@property (nonatomic, strong) IBOutlet AVPlayerViewController *av_videoPlayer;

@end

@implementation InstreamViewController

#pragma mark - Actions

- (IBAction)showSDKPlayer:(id)sender {
    
    // Remove old player.
    [self.af_videoPlayer stop];
    [self.af_videoPlayer.view removeFromSuperview];
    
    // Create the player controller with content URL.
    self.af_videoPlayer = [[AFVideoPlayerController alloc] initWithURL:self.contentURL];
    
    // Set master tags for ads.
    self.af_videoPlayer.preRollMId = kPreRollMid;
    self.af_videoPlayer.midRollMId = kMidRollMid;
    self.af_videoPlayer.postRollMId = kPostRollMid;
    
    // Add player view to view hierarchy.
    self.af_videoPlayer.view.frame = CGRectMake(0, 100, self.view.frame.size.width, self.view.frame.size.width);
    [self.view addSubview:self.af_videoPlayer.view];
    
    // Play the player.
    [self.af_videoPlayer play];
}

- (IBAction)showAVPlayer:(id)sender {
    
    // Remove old player.
    [self.af_videoPlayer stop];
    [self.af_videoPlayer.view removeFromSuperview];
    
    // Create AVPlayerViewController.
    self.av_videoPlayer = [AVPlayerViewController new];
    self.av_videoPlayer.player = [AVPlayer playerWithURL:self.contentURL];
    
    // Create content playback for player.
    AFAVPlayerViewControllerPlayback *playback = [[AFAVPlayerViewControllerPlayback alloc] initWithPlayer:self.av_videoPlayer];
    
    // Create the player controller with content playback.
    self.af_videoPlayer = [[AFVideoPlayerController alloc] initWithContainer:self.av_videoPlayer.view andContentPlayback:playback];
    
    // Set master tags for ads.
    self.af_videoPlayer.preRollMId = kPreRollMid;
    self.af_videoPlayer.midRollMId = kPreRollMid;
    self.af_videoPlayer.postRollMId = kPostRollMid;
    
    // Show the player.
    [self presentViewController:self.av_videoPlayer
                       animated:true
                     completion:nil];
}

- (NSURL *)contentURL {
    
    return [NSURL URLWithString:@"http://download.blender.org/peach/bigbuckbunny_movies/BigBuckBunny_640x360.m4v"];
}

@end

File: sample/sample/Placements/InstreamViewController.swift
//
//  InstreamViewController.swift
//  sample
//
//  Created by Jaroslav on 2022-05-30.
//

import UIKit
import AVKit
import AdformAdvertising

class InstreamViewController: UIViewController {
    
    let kPreRollMid = 1203459
    let kMidRollMid = 1203459
    let kPostRollMid = 1203459
    
    var afVideoPlayer: AFVideoPlayerController?
    var avVideoPlayer: AVPlayerViewController?
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    

    @IBAction func showAVPlayer(_ sender: Any) {
        guard let url = contentURL() else {
            return
        }
        afVideoPlayer?.stop()
        afVideoPlayer?.view.removeFromSuperview()
        
        let player = AVPlayerViewController()
        player.player = AVPlayer(url: url)
        avVideoPlayer = player
        let playBack = AFAVPlayerViewControllerPlayback(player: player)
        let afPlayer = AFVideoPlayerController(container: player.view,
                                               andContentPlayback: playBack)
        afPlayer.preRollMId = kPreRollMid
        afPlayer.midRollMId = kMidRollMid
        afPlayer.postRollMId = kPostRollMid
        afVideoPlayer = afPlayer
        self.present(player, animated: true, completion: nil)
    }
    
    @IBAction func showSDKPlayer(_ sender: Any) {
        guard let url = contentURL() else {
            return
        }
        afVideoPlayer?.stop()
        afVideoPlayer?.view.removeFromSuperview()
        
        let player = AFVideoPlayerController(url: url)
        player.preRollMId = kPreRollMid
        player.midRollMId = kMidRollMid
        player.postRollMId = kPostRollMid
        player.view.frame = CGRect(x: 0,
                                   y: 200,
                                   width: view.frame.size.width,
                                   height: view.frame.size.width)
        view.addSubview(player.view)
        afVideoPlayer = player
        afVideoPlayer?.play()
    }
    
    func contentURL() -> URL? {
        return URL(string: "http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ElephantsDream.mp4")
    }

}
